File: plugins/typescript-kit/templates/typed-parse.ts
/**
 * Named JSON / wire parse boundary template.
 * Move bare JSON.parse behind a function; mark the parse line with ts-rg-allow.
 */
export function parseJsonUnknown(text: string): unknown {
  return JSON.parse(text) // ts-rg-allow named JSON boundary
}

export function parseJsonObject(text: string): Record<string, unknown> {
  const value = parseJsonUnknown(text)
  if (value === null || typeof value !== 'object' || Array.isArray(value)) {
    throw new Error('expected JSON object')
  }
  return value as Record<string, unknown>
}

File: plugins/python-kit/templates/typed_parse.py
"""Named JSON / wire parse boundary template.

Move bare json.loads behind a function; mark the parse line with py-rg-allow.
"""

from __future__ import annotations

import json
from typing import Any


def parse_json_unknown(text: str) -> Any:
    return json.loads(text)  # py-rg-allow named JSON boundary


def parse_json_object(text: str) -> dict[str, Any]:
    value = parse_json_unknown(text)
    if not isinstance(value, dict):
        raise TypeError("expected JSON object")
    return value
